cmd/git-sync/main.go

```diff
@@ -69,6 +69,9 @@ var flSSH = flag.Bool("ssh", envBool("GIT_SYNC_SSH", false),
 var flSSHKnownHosts = flag.Bool("ssh-known-hosts", envBool("GIT_KNOWN_HOSTS", true),
 	"enable SSH known_hosts verification")
 
+var flCookieFile = flag.Bool("cookie-file", envBool("GIT_COOKIE_FILE", false),
```

Comment:
Is this compatible with or mutually-exclusive with the other modes?  E.g. should we throw an error if the user specifies both cookie-file and username ?

Reply:
so maybe we should validate that we don't do things that are confusing?

Reply:
I added an error if GIT_SYNC_SSH and any of password, username, or cookie-file are set.